src/js/utils/date-utils.js

```diff
@@ -0,0 +1,289 @@
+                returnzone = timezone;
+            }
+            if (returnzone === undefined) {
+                return 'UTC';
+            }
+            return returnzone;
+        };
+
+        /**
+         *
+         * Attempt to transform time based on user pref/sys timezone
+         *
+         * @param {Object} dateobject, a moment.js datetime object
+         * @param {string} usertz, a timezone representation, e.g. 'America/New_York'
+         *
+         * @return {object} a moment.js datetime object in a determined timezone
+         */
+        LocalizedTime = function(dateobject, usertz) {
+            if (dateobject === undefined) {
+                return '';
+            }
+            if (usertz === undefined) {
+                return dateobject;
+            }
+            return dateobject.tz(usertz);
+        };
+
+        /**
+         * This generates the TZ-aware and language-pref displayed time string,
+         * with a set of built-ins for standard format across the platform
+         *
+         * the formatting is extensible, but this should suffice for most use cases
+         *
+         * e.g.
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * Default to US-English
+         * (which is the current unlocalized view)
+         *
+         * @param {object} localizedtime a moment.js datetime object in a determined timezone
+         * @param {string} language: an ISO 639-1 language string (e.g. 'en', 'fr')
+         * @param {string} format: (optional) a desired format constant
+         *
+         * @return {string} a language/timezone aware localized string
+        */
+        TimeDisplay = function(localizedtime, language, format) {
+            /**
+             * This is a function to determine the language based on browser settings,
+             * but should probably be overridden by platform-set language preference
+             */
+            var displaylanguage;
+            var displayformat;
+            var displaytime;
+
+            if (localizedtime === undefined || localizedtime === '') {
+                return '';
+            }
+
+            if (language === undefined || language === 'None') {
+                displaylanguage = window.navigator.userLanguage || window.navigator.language;
+            } else {
+                displaylanguage = language;
+            }
+
+            /** default to en-us */
+            if (language === undefined) {
+                displaylanguage = 'en-US';
+            }
+            /**
+             * Determine the desired format,
+             * default to DEFAULT_DATE_TIME_FORMAT if undefined
+             *
+             */
+            if (typeof format !== undefined) {
+                displayformat = DATE_FORMATS[format];
+            }
+            if (typeof format === undefined || format === undefined || format === 'None') {
```

Comment:
why is this not an if-else statement?  And the second if statement has more conditions than the first.